source/agora/consensus/EnrollmentManager.d

```diff
@@ -1019,6 +1105,35 @@ unittest
     assert(man.needRevealPreimage(10));
     man.increaseNextRevealHeight();
     assert(man.needRevealPreimage(16));
+
+    // test for getting validators
+    Enrollment[] validators;
+
+    // Validator A with the `utxo_hash` and the enrolled height of 10.
+    // Validator B with the 'utxo_hash2' and the enrolled height of 11.
+    // Validator C with the 'utxo_hash3' and no enrolled height.
+    // Current block height is 1017
+    // So, valid validator are validator A and validator B
+    assert(man.updateEnrolledHeight(utxo_hash2, 11));
+    man.setCurrentBlockHeight(1017);
+    assert(man.getValidators(validators));
+    assert(validators.length == 2);
+
+    // Change the block height to 1018, which means validator A is expired.
+    man.setCurrentBlockHeight(1018);
+    assumeSafeAppend(validators);
+    validators.length = 0;
```

Comment:
`out` doesn't do what you think it does. We should have a whiteboarding session about it.

Reply:
At first, I will find and read a document for `out` and `ref`.

Reply:
`out` just means the parameter is initialized to `Type.init` when the function runs. So if you have something like this:

```D
int value = 42;
void foo (int out var)
{
   // same as var = int.init; here
    ...
}

foo(value);
```

It will initialize `value` to `int.init`, which is `0` in this case.